.changeset/big-schools-hang.md

```diff
@@ -0,0 +1,37 @@
+---
+'@xstate/store': major
+---
+
+Type parameters should now be explicitly provided to `createStore` and `createStoreWithProducer`. For sent and emitted events, the an `EventPayloadMap` should be provided, which is a map of event types to their payloads.
+
+```ts
+createStore<
+  // Context
+  {
+    count: number;
+  },
+  // Sent events
+  {
+    inc: {
+      by: number;
+    };
+  },
```

Comment:
I think the need for this explicit slot is a significant DX downgrade, but I guess you have already made up your mind on this. It feels like this now uses different set of tradeoffs than `setup` from `xstate` which is somewhat weird, even if both libraries are not strictly related.

Reply:
You don't need this if you're not using emitted events; it's still inferred. But if you are, this:

```ts
createStore({
  types: {
    emitted: {} as ...
  },
  // …
});
```

is also not really a good DX.

Reply:
I'll change this since this is now inferred